packages/discovery-react-components/src/components/StructuredQuery/__tests__/StructuredQuery.test.tsx

```diff
@@ -388,4 +467,75 @@ describe('<StructuredQuery />', () => {
       });
     });
   });
+
+  describe('On mount', () => {
+    it('Calls fetch fields', () => {
+      const mockFetchFields = jest.fn();
+      ({ structuredQuery } = setup({ apiOverrides: { fetchFields: mockFetchFields } }));
+      expect(mockFetchFields).toBeCalledTimes(1);
+      expect(mockFetchFields).toBeCalledWith();
+    });
+  });
+
+  describe('when there are fields available', () => {
+    beforeEach(() => {
+      projectFields = [{ field: 'field name 1' }, { field: 'field name 2' }];
+    });
+
+    describe('and I choose field name 1 in the fieldname combobox', () => {
+      beforeEach(() => {
+        ({ structuredQuery } = setup({ projectFields }));
+        const input = structuredQuery.getByPlaceholderText('Select field');
+        fireEvent.change(input, { target: { value: 'field name 1' } });
+      });
+
+      it('should display field name 1', () => {
+        expect(structuredQuery!.getByText('field name 1')).toBeInTheDocument();
+      });
+    });
+
+    describe('and we are fetching fields', () => {
+      beforeEach(() => {
+        fieldsStoreLoadingState = true;
+      });
+
+      describe('and I choose field name 1 in the fieldname combobox', () => {
+        beforeEach(() => {
+          ({ structuredQuery } = setup({ projectFields, fieldsStoreLoadingState }));
+          const input = structuredQuery.getByPlaceholderText('Select field');
+          fireEvent.change(input, { target: { value: 'field name 1' } });
+        });
+        it('should display field name 1', () => {
+          expect(structuredQuery!.getByText('field name 1')).toBeInTheDocument();
+        });
+      });
+    });
+  });
+
+  describe('when there are no fields available', () => {
+    beforeEach(() => {
+      projectFields = undefined;
+    });
+
+    it('should display the loading text', () => {
```

Comment:
i think it should only display the loading text if `isLoading` is `true` -> otherwise there may be the false understanding that eventually fields will be loaded even if no request is running

Reply:
I could see how this could be a better experience. Maybe I can also add an `fieldDropdownErrorText` to display if the fields store has a true value for `isError`